### DISCUSSION TITLE
How to customize shortcut keys ?

### DISCUSSION THREAD

**Discussion by hyawara** [ibhagwan/fzf-lua#1436] [Q&A] [answered]
In Telescope, use <C-j> to next when insert mode :
![image](https://github.com/user-attachments/assets/654bfdc1-9601-4455-bb84-5933fb2c0d39)

How to implement this function in fzf-lua?
I simply set up keymap:

```lua
-- insert 模式下，上下左右移动
map('i', '<C-h>', '<Left>', opt)
map('i', '<C-j>', '<Down>', opt)
map('i', '<C-k>', '<Up>', opt)
map('i', '<C-l>', '<Right>', opt)
```

However, in some plug-ins 'select ui, it cannot be moved, such as "folke/persistence.nvim":
![image](https://github.com/user-attachments/assets/bc8530b9-25d8-4ebf-96dc-76e58b6ce14f)

thanks

**Comment by cpcopy** [comment]
The default key map in fzf-lua is `<C-k>` and `<C-j>` to move between items. Are they remapped?

**Reply by hyawara** [reply]
I left the other comments out, leaving only the cmp shortcut key, but it doesn't work either:

![image](https://github.com/user-attachments/assets/071075ed-7dfd-451d-8a94-43b13b79415f)

Is it because these plug-ins rely on telescope, so I mapped them in telescope <C-j>or<C-k>, it actually took effect?

![image](https://github.com/user-attachments/assets/5bec4f81-daad-465b-a1d2-4132c7258a11)

fzf-lua is there any place to redefine the top and bottom selection list

**Reply by cpcopy** [reply]
Fzf-lua window is terminal, you may set the key map as below. Actually, it works at my side without any key map.
```
 vim.keymap.set("t", "<C-j>", "<Down>", { silent = true })
```

**Reply by ibhagwan** [reply]
Fzf defaults should have `ctrl-j|k` for up down, in case these are missing add these to your `setup` call:
```lua
  keymap = { fzf = { true, ['ctrl-j']='down', ['ctrl-k']='up' } }
```

If this doesn’t work you probably have a `<C-j>` in neovim which is overriding it, you can check with `:tmap <C-j>`.

**Reply by hyawara** [reply]
> Fzf defaults should have `ctrl-j|k` for up down, in case these are missing add these to your `setup` call:
> 
> ```lua
>   keymap = { fzf = { true, ['ctrl-j']='down', ['ctrl-k']='up' } }
> ```
> 
> If this doesn’t work you probably have a `<C-j>` in neovim which is overriding it, you can check with `:tmap <C-j>`.

I set this, but not work:

![image](https://github.com/user-attachments/assets/5c2c86eb-a871-43f1-9a51-36213e751826)

and `:tmap <C-j>` is this:
![image](https://github.com/user-attachments/assets/4ab05c4a-9450-4cc1-a424-a96dd6f58b83)

I commented out other relevant places and only kept the fzf-lua key `<C-j>` .

**Reply by hyawara** [reply]
> Fzf-lua window is terminal, you may set the key map as below. Actually, it works at my side without any key map.
> 
> ```
>  vim.keymap.set("t", "<C-j>", "<Down>", { silent = true })
> ```

For me, it still doesn't work.

**Reply by hyawara** [reply]
But if I set the telescope in insert mode<C-j>, it works, which is weird.

**Comment by hyawara** [comment]
https://github.com/user-attachments/assets/7fa1316a-5a4d-4370-99f4-c4758ab6aaf5

**Comment by ibhagwan** [comment, marked as answer]
I believe you it’s not working but you’ll have to troubleshoot this yourself, this is probably related to your terminal / tmux setup, try running `fzf` in the shell and see if ctrl-j/k works.

> But if I set the telescope in insert mode, it works, which is weird.

Not weird at all, telescope is a regular buffer, is an external process in a terminal buffer (no insert mode, terminal mode instead).

**Reply by hyawara** [reply]
I will try to solve it. Thank you very much for your help.
